### DISCUSSION TITLE
Serve static file on all routes and then invoke YARP for matched route

### DISCUSSION THREAD

**Discussion by VISSSMB** [dotnet/yarp#2402] [Q&A] [answered]
app.UseStaticFiles();
app.UseRouting();

app.MapReverseProxy();

this enables application to serve static files in wwwroot on page load and YARP is not invoked even if the path matches. Ref [#1712](https://github.com/microsoft/reverse-proxy/discussions/1712#discussioncomment-2738624) .

I have an index.html where I make a fetch of the same path such that the YARP is invoked and the response is captured in index.html.

Is there any way to achieve this same functionality for all the route patterns configured in YARP settings? I tried app.UseRewriter() but YARP seems to get invoked instead of serving index.html.

Used Order: 999 along with app.MapFallbackToFile("index.html"); and Rewriter, doesn't seem to work. Please let me know if I am missing something.

**Comment by MihaZupan** [comment, marked as answer]
> I have an index.html where I make a fetch of the same path such that the YARP is invoked and the response is captured in index.html.
> Is there any way to achieve this same functionality for all the route patterns configured in YARP settings?

I'm afraid I don't understand what you are trying to achieve. Can you try explaining in more detail / with an example?
Do you want `UseStaticFiles` to not serve some files, and instead let YARP handle the request?

**Reply by VISSSMB** [reply]
My requirement is that all routes should fallback to index.html even if YARP route matched, then again when a new fetch is made in JavaScript of index.html with the same route, YARP should be trying to forward it to destination. 

I was able to achieve this by configuring YARP to involve only when a POST request is made. All GET requests will fallback to index.html
